### PR TITLE
Ignore error when dead letter exchange cannot be found

### PR DESCRIPTION
**Pull request by dcorbacho**
[#162411672]

## Types of Changes
- [x] Bugfix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist
- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

### CODE REVIEW

File: src/rabbit_quorum_queue.erl [hunk 1]
```diff
@@ -559,9 +559,14 @@ args_policy_lookup(Name, Resolve, Q = #amqqueue{arguments = Args}) ->
...
 dead_letter_publish(X, RK, QName, ReasonMsgs) ->
-    {ok, Exchange} = rabbit_exchange:lookup(X),
-    [rabbit_dead_letter:publish(Msg, Reason, Exchange, RK, QName)
-     || {Reason, Msg} <- ReasonMsgs].
+    case rabbit_exchange:lookup(X) of
+        {ok, Exchange} ->
+            [rabbit_dead_letter:publish(Msg, Reason, Exchange, RK, QName)
+             || {Reason, Msg} <- ReasonMsgs];
+        {error, not_found} ->
+            rabbit_log:warning("Dead lettering for quorum queue ~p failed: "
```
Comment (michaelklishin) [inline src/rabbit_quorum_queue.erl]: If we log this for every message we can swamp the log very quickly.

Comment (dcorbacho) [reply src/rabbit_quorum_queue.erl]: should we ignore it then?

Comment (kjnilsson) [reply src/rabbit_quorum_queue.erl]: Probably but on the other hand this is useful information. What do classic queues do?

Comment (dcorbacho) [reply src/rabbit_quorum_queue.erl]: Nothing, they only drop the messages from the queue

**Pull request merged**